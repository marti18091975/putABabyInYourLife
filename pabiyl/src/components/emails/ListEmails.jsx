import React, { useState, useEffect } from "react";

import emailStore from "../../stores/emailStore";
import { loadEmails } from "../../actions/emailActions";
import { NavLink } from "react-router-dom";
import authStore from "../../stores/authStore";

import "./listEmails.scss";

function ListEmails(props) {
  const [emails, setEmails] = useState([]);

  const [receptorEmail] = useState(authStore.getUserEmail());

  useEffect(() => {
    emailStore.addChangeListener(onChange);

    loadEmails();

    function onChange() {
      setEmails(emailStore.getEmailsByReceptor(receptorEmail));
    }
    return () => emailStore.removeChangeListener(onChange);
  }, [receptorEmail]);
  const actualEmails = emails.map((email) => {
    const link = "/email/" + email._id;
    return (
      <NavLink className="nav-link" key={email._id} to={link}>
        <div className="email-list__item">
          <div className="emisorName email-name-text">
            from: {email.emisorName}
          </div>
          <div className="dateEmail email-text">date: {email.date}</div>

          <div className="hour email-text">
            hour: {email.hour} : {email.minute}
          </div>
        </div>
      </NavLink>
    );
  });

  return (
    <div className="main__list--email">
      {actualEmails && <ul className="email-list">{actualEmails}</ul>}
    </div>
  );
}

export default ListEmails;
