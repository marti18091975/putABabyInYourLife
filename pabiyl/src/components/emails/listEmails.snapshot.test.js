import renderer from 'react-test-renderer';
import React from 'react';
import ListEmails from './ListEmails';
import { BrowserRouter as Router } from 'react-router-dom';

describe('App snapshot', () => {

    const tree = renderer.create(
        <Router>
            <ListEmails />
        </Router>
    );

    it('should match', () => {
        expect(tree.toJSON()).toMatchSnapshot();
    });
});