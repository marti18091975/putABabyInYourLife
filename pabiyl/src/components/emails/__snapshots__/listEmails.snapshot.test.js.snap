// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should match 1`] = `
<div
  className="main__list--email"
>
  <ul
    className="email-list"
  />
</div>
`;
